hw7: functions/functional.py

```diff
@@ -0,0 +1,58 @@
+################################################################################
+# this homework explores the functionality of functional programming in Python #
+################################################################################
+
+def sequential_map(*args):
+    result = args[-1]
+    for i in args[:-1]:
+        result = list(map(i, result))
+    return result
+
+
+def consensus_filter(*args):
+    result = args[-1]
+    for i in args[:-1]:
+        result = list(filter(i, result))
+    return result
+
+
+def conditional_reduce(func1, func2, container):
+    pre_result = list(filter(func1, container))
+    result = func2(pre_result[0], pre_result[1])
+    for i in range(2, len(pre_result)):
+        result = func2(result, pre_result[2])
+    return result
```

Comment:
1. Строчка вне цикла не особо нужна
2. Каждый раз при вызове функции-редуктора ты передаёшь только элемент с индексом 2
```suggestion
    result = pre_result[0]
    for i in range(1, len(pre_result)):
        result = func2(result, pre_result[i])
    return result
```